decode_test.go

```diff
@@ -1650,3 +1650,72 @@ func (s *S) TestFuzzCrashers(c *C) {
 //		yaml.Marshal(&v)
 //	}
 //}
+
+type Figure struct {
+	Shape interface{}
+}
+
+type Circle struct {
+	Radius int
+}
+
+type Triangle struct {
+	Base int
+	Height int
+}
+
+type Figures struct {
+	Shapes []interface{}
```

Comment:
In my current use case, it would also be nice to be able to define a custom interface for e.g. Circle and Triangle and being able to use this as the slice type in the Figures struct. Example:

```
type Circle struct {...}
func (Circle) Print() {...}

type Triangle struct {...}
func (Triangle) Print() {...}

type Shape interface {
  Print()
}

type Figures struct {
  Shapes []Shape
}
```